### PR TITLE
Sample contribution

### LINKED ISSUE #361: Need to update the zenodo zarr files for 0.4.0
Currently our zendo `recOrder_test_data` for the `.zarr` still uses the old zarr structure, which is incompatible with the 0.4 ome-zarr specs implemented by iohub. For testing the new 0.4.0dev branch one needs to use the latest iohub dev and `convert()` the datasets. Will that be the expected behavior?  @ziw-liu @talonchandler 

https://zenodo.org/record/6983916/files/recOrder_test_data.zip?download=1

### PR DESCRIPTION
**Pull request by talonchandler**
Fixes #361, #370.

This .gif demonstrates the sample contributions (from local files for now) and demonstrates the birefringence overlay improvements in #426.
output2

TODO:
- [x] upload files to Zenodo
- [x] make sample contribution buttons download the files (10 MB), unzip, and open

### CODE REVIEW

**Comment by mattersoflight** [comment]
@talonchandler @ziw-liu created a zenodo entry here: https://zenodo.org/record/8280720

**Review by mattersoflight — APPROVED** [review]
Thanks @talonchandler! The code reads well to me. And the new slider to adjust the brightness of retardance is great!

Do we need to register the method for reading the sample as a reader contribution with napari?

I recall that if the user accidentally selected recOrder-napari reader, the positions in ome-zarr map to channels. Ignore this has been already taken care of separately.

**Comment by talonchandler** [comment]
Thanks for the upload @mattersoflight! 

> Do we need to register the method for reading the sample as a reader contribution with napari?
I'm currently loading the data into the viewer myself to make this as easy and as stable as possible. 

@ziw-liu I just implemented the `download_and_unzip` function we talked about. I'm storing the data in `tempfile.gettempdir()` which returns platform-specific temporary directories. My basic CI is passing across platforms and I tested by hand on Windows and macOS. Any comments on this @ziw-liu?

**Comment by ziw-liu** [comment]
> I'm storing the data in `tempfile.gettempdir()` which returns platform-specific temporary directories.

This is a clean way to store temporary files. However the files may not persist across sessions. Napari gets its built-in samples from skimage, which are stored in the application cache directories determined with [platformdirs](https://github.com/platformdirs/platformdirs).

**Comment by talonchandler** [comment]
Just changed to platformdirs...much better. Thanks @ziw-liu.

File: recOrder/scripts/samples.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import shutil
+from pathlib import Path
+
+from iohub import open_ome_zarr
+from napari.utils.notifications import show_warning
+from platformdirs import user_data_dir
+from wget import download
+
+
+def download_and_unzip():  # -> tuple(Path):
```
Comment (ziw-liu) [inline recOrder/scripts/samples.py:10]:
```suggestion
def download_and_unzip(): -> tuple(Path):
```

File: recOrder/scripts/samples.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    """Downloads sample data .zip from zenodo, unzips, and returns Paths to the .zarr datasets.
+
+    Skips the download if the files already exist.
+
+    Uses platformdirs.user_data_dir to store data.
+    """
+    temp_dirpath = Path(user_data_dir("recOrder"))
+    temp_dirpath.mkdir(exist_ok=True)
+    data_dirpath = temp_dirpath / "sample_contribution"
```
Comment (ziw-liu) [inline recOrder/scripts/samples.py:19]: Should this be version-specific so that it can be updated?

Comment (talonchandler) [reply recOrder/scripts/samples.py:19]: Good idea! Added v1.4 to folder name

File: recOrder/scripts/samples.py (hunk 1 above)
Comment (talonchandler) [reply recOrder/scripts/samples.py]: Fixed. `(` -> `[` was causing me problems

**Pull request merged**